### PR TITLE
OCPBUGS-15523: [DownstreamMerge] 6-27-23

### PR DESCRIPTION
**Pull request by trozet**
No conflicts

@dcbw @jordigilh @tssurya @flavio-fernandes @kyrtapz fyi

### CODE REVIEW

**Comment by trozet** [comment]
/hold

we are probably going to want:
https://github.com/ovn-org/ovn-kubernetes/pull/3714

The crash was seen in this PR

**Comment by tssurya** [comment]
haha the payloads passed so beautifully... but yea let's get that panic in and also the bits that can stop the upgrade container restarts from happening.

**Comment by tssurya** [comment]
/lgtm
I know the payload jobs don't look that green but some of them are green, the ones that are red.. can't spot ovnk issues from a first glance.
I am adding the lgtm but will let @trozet remove the hold when he thinks its ok.

File: go-controller/pkg/ovn/master.go [hunk 1]
```diff
@@ -602,11 +603,46 @@ func (oc *DefaultNetworkController) syncNodes(kNodes []interface{}) error {
 	if err != nil {
 		return fmt.Errorf("failed to get node logical switches which have other-config set: %v", err)
 	}
+
+	staleNodes := sets.NewString()
 	for _, nodeSwitch := range nodeSwitches {
 		if !foundNodes.Has(nodeSwitch.Name) {
```
Comment (flavio-fernandes) [inline go-controller/pkg/ovn/master.go]: Note: this logic is not taking transit switch into account. that has been addressed via commit https://github.com/openshift/ovn-kubernetes/pull/1726/commits/e832edec6eda073afe407a3bb31efbb30cc614f0 which is also part of this PR.

**Review by flavio-fernandes — APPROVED** [review]
Took at the commits related to changes I have recently been working on.
/lgtm

**Pull request merged**